examples/c/publish_subscribe_cross_language/src/publisher.c

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+#include "iox2/iceoryx2.h"
+#include "message_data.h"
+
+#ifdef _WIN64
+#define alignof __alignof
+#else
+#include <stdalign.h>
+#endif
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+
+int main(void) {
+    // create new node
+    iox2_node_builder_h node_builder_handle = iox2_node_builder_new(NULL);
+    iox2_node_h node_handle = NULL;
+    if (iox2_node_builder_create(node_builder_handle, NULL, iox2_service_type_e_IPC, &node_handle) != IOX2_OK) {
+        printf("Could not create node!\n");
+        goto end;
+    }
+
+    // create service name
+    const char* service_name_value = "My/Funk/ServiceName";
+    iox2_service_name_h service_name = NULL;
+    if (iox2_service_name_new(NULL, service_name_value, strlen(service_name_value), &service_name) != IOX2_OK) {
+        printf("Unable to create service name!\n");
+        goto drop_node;
+    }
+
+    // create service builder
+    iox2_service_name_ptr service_name_ptr = iox2_cast_service_name_ptr(service_name);
+    iox2_service_builder_h service_builder = iox2_node_service_builder(&node_handle, NULL, service_name_ptr);
+    iox2_service_builder_pub_sub_h service_builder_pub_sub = iox2_service_builder_pub_sub(service_builder);
+
+    // set pub sub payload type
+    // payload_type_name is equivalent to the payload type name used on the Rust side and was determined with
+    // `core::any::type_name::<TransmissionData>()`
+    const char* payload_type_name = "examples_common::transmission_data::TransmissionData";
```

Comment:
Can we move this and the `user_header_type_name` into `message_data.h` (since it shall always tightly coupled to the type definition) and turn it into a macro? `#define USER_HEADER_TYPE_NAME "examples..."`